src/Sarif.ValidationTests/SarifValidatorTests.cs

```diff
@@ -195,5 +207,46 @@ private string FailureReason(Result[] errors)
 
             return sb.ToString();
         }
+
+        private static bool RecursivePropertySearch(JObject parentObject, string searchKey)
```

Comment:
>string searchKey [](start = 74, length = 16)

i think i should change this to accept string[] so that we can test for other fields like rank as well - i.e. the ones which should not be serialized.